### PR TITLE
Fix typos under torch/distributed directory

### PR DESCRIPTION
**Pull request by kiszk**
This PR fixes typos in comments and messages of `.py` files under `torch/distributed` directory

### CODE REVIEW

**Review by soulitzer — APPROVED** [review]

File: torch/distributed/fsdp/flat_param.py [hunk 1]
```diff
@@ -1943,7 +1943,7 @@ def _writeback_orig_params(self) -> bool:
                         flat_param_grad.untyped_storage().data_ptr()
                     )
         # TODO: If we want to handle shared parameters, we need to re-generate
-        # the shared parameter data structures in case sharedness changed.
+        # the shared parameter data structures in case shardness changed.
```
Comment (awgu) [inline torch/distributed/fsdp/flat_param.py:1946]: This should be sharedness (even if that is not a real word). I meant the state of being "shared".

Comment (kiszk) [reply torch/distributed/fsdp/flat_param.py]: @awgu thank you for your commented. I updated the branch

Comment (awgu) [reply torch/distributed/fsdp/flat_param.py]: Thanks!

**Review by kit1980 — APPROVED** [review]

**Pull request closed without merging**